vscode-lean4/src/utils/projectInfo.ts

```diff
@@ -0,0 +1,146 @@
+import * as fs from 'fs';
+import { URL } from 'url';
+import { Uri, workspace, WorkspaceFolder } from 'vscode';
+
+// Find the root of a Lean project and return the Uri for the package root and the Uri
+// for the 'leanpkg.toml' or 'lean-toolchain' file found there.
+export function findLeanPackageRoot(uri: Uri) : [WorkspaceFolder, Uri,Uri] {
+    if (!uri) return [null, null, null];
```

Comment:
```suggestion
export function findLeanPackageRoot(uri: Uri) : [WorkspaceFolder | null, Uri | null, Uri | null] {
    if (!uri) return [null, null, null];
```

I'm surprised that this passes the type checker.  Another more self-explanatory return type might be `{workspaceFolder?: WorkspaceFolder, leanpkgToml?: Uri, leanToolchain?: Uri}`.